packages/react-composites/src/composites/MeetingComposite/ChatButtonLogicWrapper.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ChatMessage } from '@azure/communication-chat';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { ChatAdapter } from '../ChatComposite';
+import { ChatButton, ChatButtonProps } from './ChatButton';
+
+/**
+ * @private
+ */
+export type ChatButtonLogicWrapperProps = {
+  chatButtonProps: ChatButtonProps;
+  chatAdapter: ChatAdapter;
+  showChat: boolean;
+  onChatButtonClicked: () => void;
+  disabled: boolean;
+};
+
+/**
+ * @private
+ */
+export const ChatButtonLogicWrapper = (props: ChatButtonLogicWrapperProps): JSX.Element => {
```

Comment:
`LogicWrapper` could mean anything, name this something like `ChatButtonWithUnreadMessagesBadge` (or something less verbose if you can think of it)